controllers/provisioning_controller.go

```diff
@@ -630,14 +618,24 @@ func (r *ProvisioningReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	klog.InfoS("Network stack calculation", "NetworkStack", r.NetworkStack)
 
+	provisioningFilter := predicate.NewPredicateFuncs(func(object client.Object) bool {
+		if object.GetName() == metal3iov1alpha1.ProvisioningSingletonName {
+			return true
+		}
+		klog.Info("ignoring provisioning config with unexpected name", "name", object.GetName(), "expected-name", metal3iov1alpha1.ProvisioningSingletonName)
+		return false
+	})
+	clusterOperatorFilter := predicate.NewPredicateFuncs(func(object client.Object) bool { return object.GetName() == clusterOperatorName })
+
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&metal3iov1alpha1.Provisioning{}).
+		For(&metal3iov1alpha1.Provisioning{}, builder.WithPredicates(provisioningFilter)).
 		Owns(&corev1.Secret{}).
 		Owns(&appsv1.Deployment{}).
 		Owns(&corev1.Service{}).
 		Owns(&appsv1.DaemonSet{}).
 		Owns(&osconfigv1.ClusterOperator{}).
 		Owns(&osconfigv1.Proxy{}).
 		Owns(&machinev1beta1.Machine{}).
+		Watches(&source.Kind{Type: &osconfigv1.ClusterOperator{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(clusterOperatorFilter)).
```

Comment:
@sanchezl Does this do anything?
AIUI `EnqueueRequestForObject{}` makes the controller reconcile the Provisioning CR with the same name and namespace as the ClusterOperator that changed. Which of course doesn't exist, hence why you had to add a predicate to filter it out again, leaving us exactly where we started.
This appears to me to be just as ineffective as our previous attempt to watch the ClusterOperator on line 636, which has no effect because the Provisioning CR does not own any ClusterOperator CRs (and nor should it).

Reply:
https://issues.redhat.com/browse/OCPBUGS-36357